main/src/com/google/refine/importers/SeparatorBasedImporter.java

```diff
@@ -136,21 +142,28 @@ public void parseOneFile(
             quote = CharMatcher.whitespace().trimFrom(quoteCharacter).charAt(0);
         }
 
-        CsvParserSettings settings = new CsvParserSettings();
-        CsvFormat format = settings.getFormat();
-        format.setDelimiter(sep);
-        format.setQuote(quote);
-        format.setLineSeparator("\n");
-        settings.setIgnoreLeadingWhitespaces(false);
-        settings.setIgnoreTrailingWhitespaces(false);
-        if (strictQuotes) {
-            settings.setUnescapedQuoteHandling(UnescapedQuoteHandling.RAISE_ERROR);
+        AbstractParser parser;
+        if (tsv) {
+            TsvParserSettings settings = new TsvParserSettings();
+            settings.setMaxCharsPerColumn(256 * 1024); // TODO: Perhaps use a lower default and make user configurable?
```

Comment:
I think it makes sense to have different default settings for TSV vs CSV but with this change we'd be ignoring quote settings (the quote character, and the setting to preserve leading/trailing whitespace), as soon as the user uses tabs as separator, right?
As a user I'd find it confusing to see that options I am offered aren't actually honored.

Would it not be preferable to integrate this logic in `createParserUIInitializationData`, so that the default options for those parameters are set to sensible values, and then we can stick to passing them to `CsvFormat` regardless of the separator, as it used to be?

Reply:
Ok, I see the options are honored, thanks for the clarification. I think the UI is fine as such.